components/script/dom/fakexrdevice.rs

```diff
@@ -0,0 +1,142 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+use crate::dom::bindings::codegen::Bindings::FakeXRDeviceBinding::{
+    self, FakeXRDeviceMethods, FakeXRRigidTransformInit, FakeXRViewInit,
+};
+use crate::dom::bindings::codegen::Bindings::XRViewBinding::XREye;
+use crate::dom::bindings::error::{Error, Fallible};
+use crate::dom::bindings::reflector::{reflect_dom_object, DomObject, Reflector};
+use crate::dom::bindings::root::DomRoot;
+use crate::dom::globalscope::GlobalScope;
+use dom_struct::dom_struct;
+use webvr_traits::{MockVRControlMsg, MockVRView, WebVRMsg};
+
+#[dom_struct]
+pub struct FakeXRDevice {
+    reflector: Reflector,
+}
+
+impl FakeXRDevice {
+    pub fn new_inherited() -> FakeXRDevice {
+        FakeXRDevice {
+            reflector: Reflector::new(),
+        }
+    }
+
+    pub fn new(global: &GlobalScope) -> DomRoot<FakeXRDevice> {
+        reflect_dom_object(
+            Box::new(FakeXRDevice::new_inherited()),
+            global,
+            FakeXRDeviceBinding::Wrap,
+        )
+    }
+
+    pub fn send_msg(&self, msg: MockVRControlMsg) {
+        self.global()
+            .as_window()
+            .webvr_thread()
+            .unwrap()
```

Comment:
Can we not panic here, especially since this is a public method?

Reply:
It's private now, and you shouldn't be running XR tests without the webvr thread existing.